app/Jobs/GeneratePackageOpenGraphImage.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace App\Jobs;
+
+use GDText\Box;
+use GDText\Color;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Storage;
+
+class GeneratePackageOpenGraphImage implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    protected $packageName;
+    protected $packageAuthor;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(String $packageName, String $packageAuthor)
+    {
+        $this->packageName = $packageName;
+        $this->packageAuthor = $packageAuthor;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
```

Comment:
I'd pull each of the responsibilities of this job at *least* into its own method here. Maybe even separate classes, but let's start with their own methods.

So, what does this method do?

1. Ensures a directory exists
2. Checks if the file already exists
3. Does a crap ton of image generate-y stuff
4. Saves the file

I'd love to have at *least* a separate method (maybe a class but I'm not sure) which has three params: title, name, and og path. This method generates the image and saves it to that path, without any of the other logic in here.